plugins/microsoft_intune/icon_microsoft_intune/util/api.py

```diff
@@ -214,3 +304,15 @@ def create_necessary_headers(access_token):
             "Authorization": f"Bearer {access_token}",
             "Content-Type": "application/json",
         }
+
+    def _filter_policy_id(self, managed_app_policies, policy_name):
+        for item in managed_app_policies["value"]:
+            if item["displayName"] == policy_name:
+                return item["id"]
+        return None
+
+    def _filter_app_package_id(self, managed_app_list, application_name):
+        for item in managed_app_list["content"]["appList"]:
```

Comment:
We should also verify that this structure is consistent and we can hardcode these as constants. In which case, we should use programatic constants.